src/Sarif.WorkItems/SarifWorkItemFiler.cs

```diff
@@ -290,6 +290,12 @@ public SarifWorkItemModel FileWorkItemInternal(SarifLog sarifLog, SarifWorkItemC
             {
                 string logId = sarifLog.GetProperty<Guid>(LOGID_PROPERTY_NAME).ToString();
 
+                if (sarifLog.Runs?.Any(run => run.Results?.Any(result => result.ShouldBeFiled()) == true) != true)
```

Comment:
do we think also add a safety check in this method as well

public static bool ShouldBeFiled(this Result result)
{
    if(result ==null) return false;
    ...... 
}

Reply:
Added the check to ShouldBeFiled() and added test cases.